larq/quantizers.py

```diff
@@ -532,6 +533,60 @@ def grad(dy):
     return _and_binarize_with_identity_grad(x)
 
 
+def _dorefa_k_bit_quantizer(x, k_bit=2):
+    x = tf.clip_by_value(x, 0.0, 1.0)
+
+    @tf.custom_gradient
+    def _k_bit_with_identity_grad(x):
+        n = 2 ** k_bit - 1
+        return tf.round(x * n) / n, lambda dy: dy
+
+    return _k_bit_with_identity_grad(x)
+
+
+@utils.register_keras_custom_object
+class DorefaQuantizer(QuantizerFunctionWrapper):
+    r"""Instantiates a serializable k_bit quantizer as in the DoReFa paper.
+
+    \\[
+    q(x) = \begin{cases}
+    0 & x < \frac{1}{2n} \\\
+    \frac{i}{n} & \frac{2i-1}{2n} < |x| < \frac{2i+1}{2n} \text{ for } i \in \\{1,n-1\\}\\\
+     1 & \frac{2n-1}{2n} < x
+    \end{cases}
+    \\]
+
+    where \\(n = 2^{\text{k_bit}} - 1\\). The number of bits, k_bit, needs to be passed as an argument.
+    The gradient is estimated using the Straight-Through Estimator
+    (essentially the binarization is replaced by a clipped identity on the
+    backward pass).
+    \\[\frac{\partial q(x)}{\partial x} = \begin{cases}
+    1 &  0 \leq x \leq 1 \\\
+    0 & \text{else}
+    \end{cases}\\]
+    Note that in the original tensorpack implementation the clip and
+    the quantizer are two different functions that get composed.
```

Comment:
```suggestion
```

Reply:
I guess you suggest to delete the whole sentence right?